File: user-service/src/main/java/com/chong/pay/userservice/domain/PayUser.java
package com.chong.pay.userservice.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document
public class PayUser {
    public enum ChargeMethod {
        CARD, BANK_ACCOUNT
    }

    @Id
    private String userId;
    private String email;
    private Set<ChargeMethod> chargeMethods;
    private Set<String> cardCompanyNames;
    private Set<String> bankCompanyNames;
    private long payMoney;
    private long point;
}

File: transfer-service/src/main/kotlin/com/chong/pay/transferservice/domain/PayUser.kt
package com.chong.pay.transferservice.domain

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

enum class ChargeMethod {
    CARD, BANK_ACCOUNT
}

@Document
data class PayUser (
    @Id
    var userId: String,
    var email: String? = null,
    var chargeMethods: Set<ChargeMethod>? = null,
    var cardCompanyNames: Set<String>? = null,
    var bankCompanyNames: Set<String>? = null,
    var payMoney: Long,
    var point: Long? = null
)